### ISSUE TITLE
Missing pages

### ISSUE THREAD

**Issue by zimonh** [smalot/pdfparser#471]
when parsing [this](https://drive.google.com/file/d/1oawNtAdqoFZ29AnA0iatTDWdYmUCFLnb/view?usp=sharing) folder the number of pages (25) does not match the actual page number (37).

**Comment by k00ni** [comment]
In case someone wanna work on this: Can we include linked PDF in our test environment? It must be free of charge and without any obligations.

**Comment by zimonh** [comment]
Yes it's a digital promo folder for a dutch supermarket. thx for the speedy response.

**Comment by izabala** [comment]
I took a look at this.  I couldn't find the problem. I didn't catch where is the error.  I only know that after the method `Parser::parseContent` is called, and exactly after the line:` $document->setObjects($this->objets);` of that method; is when  you start seeing 25 pages instead of the 37.  But I am not skillful enough in the `Parse::ParseContent` to understand how the raw data that is in `$data` variable (after the call of `list($xref, $data) = $this->rawDataParser->parseData($content);`) could be interpreted to know what is a page or not, and what is the problem.

I run a example, just for see if whit other php library, I have the same problem with page number.  I used FPDI to get the page number, but as soon as FPDI get the file, php throw the following Fatal error:

```
PHP Fatal error:  Uncaught setasign\Fpdi\PdfParser\CrossReference\CrossReferenceException: This PDF document probably uses
 a compression technique which is not supported by the free parser shipped with FPDI. 
(See https://www.setasign.com/fpdi-pdf-parser for more details) in vendor\setasign\fpdi\src\PdfParser\CrossReference\CrossReference.php:257
```

I just include this, to see if this could help others to find the problem.

Sorry, I can't help any more, I am not expert in the method:  `Parse::ParseContent` and related methods.

**Comment by k00ni** [comment]
> Do you have the knowledge to help me in the final stretch here?

Thank you for investing time in resolving this issue. My suggestion to you is to create a pull request, outline your thought and provide your code changes. There we can discuss them and try to find a solution. I am currently busy, but if I can I would be happy to support you (and others involved) here.
